project-base/src/Shopsys/ShopBundle/DataFixtures/Demo/ProductDataFixture.php

```diff
@@ -4,129 +4,5838 @@
+        $productData->usingStock = true;
+        $productData->stockQuantity = 140;
+        $productData->outOfStockAction = Product::OUT_OF_STOCK_ACTION_HIDE;
+
+        $this->setUnit($productData, UnitDataFixture::UNIT_PIECES);
+        $this->setAvailability($productData, AvailabilityDataFixture::AVAILABILITY_IN_STOCK);
+        $this->setCategoriesForAllDomains($productData, [CategoryDataFixture::CATEGORY_COFFEE]);
+        $this->setFlags($productData, [FlagDataFixture::FLAG_NEW_PRODUCT]);
+
+        $productData->sellingDenied = false;
+        $this->setBrand($productData, BrandDataFixture::BRAND_DELONGHI);
+
+        $this->createProduct($productData);
+
+        $productData = $this->productDataFactory->create();
+
+        $productData->catnum = '5962199';
+        $productData->partno = '';
+        $productData->ean = '8845781245958';
+
+        foreach ($this->domain->getAllIncludingDomainConfigsWithoutDataCreated() as $domain) {
+            $locale = $domain->getLocale();
+            $productData->name[$locale] = t('Pot holder, black', [], 'dataFixtures', $locale);
+            $productData->descriptions[$domain->getId()] = t('This pot holder is used to hold pots. No more burnt kitchen tables!', [], 'dataFixtures', $domain->getLocale());
+            $productData->shortDescriptions[$domain->getId()] = t('This pot holder is used to hold pots. No more burnt kitchen tables!', [], 'dataFixtures', $domain->getLocale());
+        }
+
+        $this->setPriceForAllPricingGroups($productData, '3');
+
+        $this->setVat($productData, VatDataFixture::VAT_HIGH);
+        $this->setSellingFrom($productData, '13.2.2014');
+        $this->setSellingTo($productData, null);
+        $productData->usingStock = false;
+        $productData->stockQuantity = 0;
+        $productData->outOfStockAction = Product::OUT_OF_STOCK_ACTION_HIDE;
+
+        $this->setUnit($productData, UnitDataFixture::UNIT_PIECES);
+        $this->setAvailability($productData, AvailabilityDataFixture::AVAILABILITY_OUT_OF_STOCK);
+        $this->setCategoriesForAllDomains($productData, [CategoryDataFixture::CATEGORY_BOOKS]);
+        $this->setFlags($productData, [FlagDataFixture::FLAG_NEW_PRODUCT]);
+
+        $productData->sellingDenied = true;
+        $this->setBrand($productData, null);
+
+        $this->createProduct($productData);
+
+        $productData = $this->productDataFactory->create();
+
+        $productData->catnum = '5960139';
+        $productData->partno = '31011039100';
+        $productData->ean = '8845781245924';
+
+        $parameterTranslations = [];
+
+        foreach ($this->domain->getAllIncludingDomainConfigsWithoutDataCreated() as $domain) {
+            $locale = $domain->getLocale();
+            $productData->name[$locale] = t('Genius NetScroll 310 silver', [], 'dataFixtures', $locale);
+            $productData->descriptions[$domain->getId()] = t('Simple Genius Mouse NetScroll 310 offers the most widely used functional elements of 
```

Comment:
Passing by reference is a poor design. Will easily become unpredictable in such a large class.

Reply:
I don't really see a better solution that would be in scope of this US. What do you suggest?

Reply:
At least I would create a method for creating parameters. Doing so will remove iterator `$i` and then you may obtain `$parameterTranslations` by the `return`.

And do not know why, byt marked lines are wrong. The usage I was pointing to is e.g.
```
            $this->addParameterTranslations($parameterTranslations, t('Technology', [], 'dataFixtures', $locale), t('Optical', [], 'dataFixtures', $locale), $locale, $i);
            $this->addParameterTranslations($parameterTranslations, t('Interface', [], 'dataFixtures', $locale), t('Wired', [], 'dataFixtures', $locale), $locale, $i);
```

Reply:
I am sorry, I am closing this because of time consumption vs benefit